Documentation/contributors/design-proposals/webhooks.md

```diff
@@ -0,0 +1,213 @@
+
+### CSV example that includes webhook configuration (v2)
+
+This example serves as a potential method of CSV configuration that gives minimal control to the user. Portions of configuration that have been eliminated are commented out in order to make an easier comparison.
+
+  install:
+    strategy: deployment
+    spec:
+      clusterPermissions:
+        - serviceAccountName: my-operator
+          rules:
+          - apiGroups:
+            - "*"
+            ...
+      permissions:
+        - serviceAccountName: my-operator
+          rules:
+          - apiGroups:
+            - ""
+            ...
+      deployments:
+        - name: webhook-deployment
+          spec:
+            replicas: 1
+            selector:
+              matchLabels:
+                name: mywebhook
+            template:
+              metadata:
+                labels:
+                  name: mywebhook
+              spec:
+                serviceAccountName: my-operator
+                containers:
+                - name: webhook
+                  image: operator-framework/fake-webhook-image:latest
+                  imagePullPolicy: IfNotPresent
+                  ports:
+                  - containerPort: 443
+                  env:
+                  - name: NAMESPACE
+                    valueFrom:
+                      fieldRef:
+                        fieldPath: metadata.annotations['olm.targetNamespaces']
+                  #volumeMounts:
+                  #  - mountPath: /keys
+                  #    name: tls-keys
+                  #    readonly: true
+                - name: operator
+                  image: operator-framework/fake-operator:latest
+                  imagePullPolicy: IfNotPresent
+                  env:
+                  - name: NAMESPACE
+                    valueFrom:
+                      fieldRef:
+                        fieldPath: metadata.annotations['olm.targetNamespaces']
+                  - name: MY_POD_NAME
+                    valueFrom:
+                      fieldRef:
+                        fieldPath: metadata.name
+                #volumes:
+                #- name: tls-keys
+                #    secretName: service-tls
+      #services:
+      #- name: mywebhook-service
+      #  namespace: test-ns
+      #  spec:
+      #    selector:
+      #      name: mywebhook
+      #    ports:
+      #    - protocol: TCP
+      #      port 443
+      #      targetPort: 443
+      validatingWebhooks:
+      - name: olm-admission
+        webhooks:
+          - name: csv.operators.olm.com
+          rules:
+            - apiGroups:
+              - "operators.olm.com"
+              apiVersions:
+              - v1alpha1
+              operations:
+              - CREATE
+              resources:
+              - clusterserviceversions
```

Comment:
In the absence of the client config, I think we need to also take a pod selector, so that we can generate the service correctly. right?